### PR TITLE
Library tour example tests for dart:io, dart:convert, dart:mirrors

### PR DESCRIPTION
**Pull request by chalin**
Contributes to #407 and #416, wrapping up all the rest of the library tour examples.

Continuation of #443

Staged at https://dartlang-org-staging-0.firebaseapp.com/guides/libraries/library-tour#dartio---io-for-command-line-apps

### CODE REVIEW

File: src/_guides/libraries/library-tour.md [hunk 1]
```diff
@@ -2576,20 +2548,20 @@ The first parameter specifies the method to be invoked, and the second
...
 parameter lets you specify named arguments.
 
-<!-- library-tour/mirrors/bin/main.dart -->
+<?code-excerpt "test/library_tour/mirrors_test.dart (invoke)"?>
 {% prettify dart %}
 var p = new Person('Bob', 'Smith', 42);
 InstanceMirror mirror = reflect(p);
 
-mirror.invoke(#greet, ['Shailen']);
+mirror.invoke(#greet, ['Sundar']);
```
Comment (chalin) [inline src/_guides/libraries/library-tour.md]: FYI, this change was copied from https://github.com/dart-lang/site-www/blob/792c517504acffcbffd07f85297fc43b134581ea/examples/library_tour/mirrors/bin/main.dart#L80-L81

File: src/_guides/libraries/library-tour.md [hunk 2]
```diff
@@ -2487,68 +2459,68 @@ To begin, you need to *reflect* on a class or object to get its
...
-    print('The field ${m.simpleName} is $privateStatus ' +
-          'and $finalStatus and is annotated as ' +
-          '$typeAnnotation.');
+    print('The field ${v.simpleName} is $privateStatus ' +
+        'and $finalStatus and is annotated as ' +
+        '$typeAnnotation.');
   });
-}{% endprettify %}
+}
+{% endprettify %}
```
Comment (chalin) [inline src/_guides/libraries/library-tour.md]: FYI, the updates from the code example above are from https://github.com/dart-lang/site-www/blob/master/examples/library_tour/mirrors/bin/main.dart

File: src/_guides/libraries/library-tour.md [hunk 3]
```diff
@@ -2333,61 +2316,49 @@ the object's `toJson()` method.
...
-import 'dart:convert' show UTF8;
-
-main() {
-  List<int> expected = [
-    0xc3, 0x8e, 0xc3, 0xb1, 0xc5, 0xa3, 0xc3, 0xa9,
-    0x72, 0xc3, 0xb1, 0xc3, 0xa5, 0xc5, 0xa3, 0xc3,
-    0xae, 0xc3, 0xb6, 0xc3, 0xb1, 0xc3, 0xa5, 0xc4,
-    0xbc, 0xc3, 0xae, 0xc5, 0xbe, 0xc3, 0xa5, 0xc5,
-    0xa3, 0xc3, 0xae, 0xe1, 0xbb, 0x9d, 0xc3, 0xb1
-  ];
```
Comment (chalin) [inline src/_guides/libraries/library-tour.md]: No need to repeat this long array. In the page it is displayed just a few lines above in the previous excerpt.

File: src/_guides/libraries/library-tour.md [hunk 4]
```diff
@@ -2407,6 +2378,11 @@ invoke functions or methods at runtime.
...
 To use it, import dart:mirrors.
 
+<?code-excerpt "test/library_tour/mirrors_test.dart (import)"?>
+{% prettify dart %}
+import 'dart:mirrors';
+{% endprettify %}
+
 <div class="alert alert-warning" markdown="1">
 **Warning:**
 Using dart:mirrors can cause dart2js to generate very large JavaScript
```
Comment (filiph) [inline src/_guides/libraries/library-tour.md]: I wonder if we are ready to update this section.  Maybe keep mirrors & `@MirrorsUsed` but also add a paragraph about `package:build` etc. Not necessarily in this PR.

cc @kevmoo to see if we're ready for this. Newcomers to Dart still learn about mirrors before anything else.

File: examples/test/language_tour/template_test.dart [hunk 5]
```diff
@@ -1,8 +1,9 @@
 // ignore_for_file: unused_element, type_annotate_public_apis
 // #docplaster
-import 'package:test/test.dart';
+// import 'package:test/test.dart';
```
[the remark this answers is not in the document]
Comment (chalin) [reply examples/test/language_tour/template_test.dart]: It's a template file I've been using when creating new test files. I've removed it.

File: src/_guides/libraries/library-tour.md [hunk 6]
```diff
@@ -1982,8 +1978,13 @@ Dart has additional libraries for more specialized web APIs, such as
 
 The [dart:io library][dart:io] provides APIs to
 deal with files, directories, processes, sockets, WebSockets, and HTTP
-clients and servers. Only command-line scripts and servers can use
-dart:io—not web apps.
+clients and servers.
+
+<div class="alert alert-warning" markdown="1">
+  **Important:**
+  Only command-line scripts and servers can import and use `dart:io`,
```
[the remark this answers is not in the document]
Comment (chalin) [reply src/_guides/libraries/library-tour.md]: Added "Flutter mobile apps" to the "only" list.

**Comment by chalin** [comment]
Thanks for the feedback @filiph

**Pull request merged**